### PR TITLE
Add Advice and Advise functions for adding an advice section to any span.

### PR DESCRIPTION
**Pull request by sgmiller**
Also finishes implementing indentation and adds word wrapping to messages,
warnings, and advice.

### CODE REVIEW

File: command/operator_diagnose.go [hunk 1]
```diff
@@ -176,7 +176,12 @@ func (c *OperatorDiagnoseCommand) RunWithParsedFlags() int {
 		c.UI.Output(string(resultsJS))
 	} else {
 		c.UI.Output("\nResults:")
-		results.Write(os.Stdout)
+		w, _, err := term.GetSize(0)
```
Comment (swayne275) [inline command/operator_diagnose.go:179]: i'm not sure what's going on here.

i looked up the doc for `term.GetSize()` and it looks like it gives the width, height, and error of the current terminal window. I'm not sure what the `0` input means, or why we'd want to return the width?

File: vault/diagnose/output.go [hunk 2]
```diff
@@ -310,62 +273,107 @@ func (t *TelemetryCollector) getOrBuildResult(id trace.SpanID) *Result {
 	return r
 }
 
+func findAttributes(e trace.Event, attr1, attr2 attribute.Key) (string, string) {
+	var av1, av2 string
```
Comment (swayne275) [inline vault/diagnose/output.go:277]: why are these called `av1` and `av2`?

File: vault/diagnose/output.go [hunk 3]
```diff
@@ -310,62 +273,107 @@ func (t *TelemetryCollector) getOrBuildResult(id trace.SpanID) *Result {
...
-			sb.WriteString(status_failed)
+			prelude = status_failed
 		case SkippedStatus:
-			sb.WriteString(status_skipped)
+			prelude = status_skipped
 		}
-		sb.WriteString(r.Name)
+		prelude = prelude + r.Name
 
 		if r.Message != "" || len(r.Warnings) > 0 {
```
Comment (swayne275) [inline vault/diagnose/output.go:334]: why do we check message or warnings, but then only include the message?

**Review by swayne275 — COMMENTED** [review]
looks pretty good overall

File: command/operator_diagnose.go (hunk 1 above)
Comment (sgmiller) [reply command/operator_diagnose.go:179]: Yeah, we're trying to determine the with of the terminal we're running in.  The 0 is the file descriptor for "stdout", which identifies the correct TTY for the terminal.  We're only interested in width, which is what we pass to set the wordwrap limit.

File: vault/diagnose/output.go (hunk 2 above)
Comment (sgmiller) [reply vault/diagnose/output.go:277]: attribute value 1 and 2.  This function is generically grabbing up two two attribute values given their keys.

File: command/operator_diagnose.go (hunk 1 above)
Comment (swayne275) [reply command/operator_diagnose.go:179]: ohhhhh totally makes sense now. thanks!

File: vault/diagnose/output.go (hunk 3 above)
Comment (sgmiller) [reply vault/diagnose/output.go:334]: yeah, that's not quite right.  The intent is to output the message if present, or the first warning if there are warnings but no message.  But this is outputting a blank message if there is no message but some warnings.  Will fix.

**Review by swayne275 — APPROVED** [review]

**Pull request merged**